js/src/Ice/ObjectPrxExtensions.js

```diff
@@ -501,11 +501,7 @@ ObjectPrx.checkedCast = function (prx, facet, ctx) {
                 r.resolve(ret ? new this(prx) : null);
             })
             .catch((ex) => {
-                if (ex instanceof FacetNotExistException) {
-                    r.resolve(null);
-                } else {
-                    r.reject(ex);
-                }
+                r.reject(ex);
```

Comment:
Is this still necessary?
My  understanding is that `Promise::catch` is only called for `reject`ed promises anyways.
So this feels alot like a pointless `catch(ex) { throw ex; }`

Reply:
we can write it as:

```
prx.ice_isA(this.ice_staticId(), ctx).then(
            (ret) => r.resolve(ret ? new this(prx) : null),
            (ex) => r.reject(ex),
        );
```

Reply:
Ah yes, because `r` is a different promise than the one being returned by the `ice_isA` operation. I'd missed that before.

Reply:
Updated code using Jose's suggestion, except without parenthesis around ret and ex.